File: AI-Services/app/utils/report_logger.py
# app/utils/report_logger.py

import os
from datetime import datetime

REPORT_DIR = "reports"

if not os.path.exists(REPORT_DIR):
    os.makedirs(REPORT_DIR)

def create_report_file(prefix="ai-report"):
    timestamp = datetime.now().isoformat().replace(":", "-")
    filename = f"{prefix}-{timestamp}.txt"
    path = os.path.join(REPORT_DIR, filename)

    with open(path, "w") as f:
        f.write("")

    return path

def append_report(path, text):
    with open(path, "a", encoding="utf-8") as f:
        f.write(text + "\n")

File: Backend/src/utils/reportLogger.ts
import fs from "fs";
import path from "path";

const REPORT_DIR = path.join(process.cwd(), "reports");

// ensure folder exists
if (!fs.existsSync(REPORT_DIR)) {
  fs.mkdirSync(REPORT_DIR);
}

export const createReportFile = (prefix: string) => {
  const timestamp = new Date().toISOString().replace(/[:.]/g, "-");
  const filename = `${prefix}-${timestamp}.txt`;
  const filePath = path.join(REPORT_DIR, filename);

  fs.writeFileSync(filePath, ""); // create empty file

  return filePath;
};

export const appendToReport = (filePath: string, content: string) => {
  fs.appendFileSync(filePath, content + "\n");
};